discovery/kubernetes/service_test.go

```diff
@@ -254,3 +254,87 @@ func TestServiceDiscoveryNamespaces(t *testing.T) {
 		},
 	}.Run(t)
 }
+
+func TestServiceDiscoveryOwnNamespace(t *testing.T) {
+	n, c := makeDiscovery(RoleService, NamespaceDiscovery{IncludeOwnNamespace: true})
+
+	k8sDiscoveryTest{
+		discovery: n,
+		afterStart: func() {
+			for _, ns := range []string{"own-ns", "non-own-ns"} {
+				obj := makeService()
+				obj.Namespace = ns
+				c.CoreV1().Services(obj.Namespace).Create(context.Background(), obj, metav1.CreateOptions{})
+			}
+		},
+		expectedMaxItems: 1,
+		expectedRes: map[string]*targetgroup.Group{
+			"svc/own-ns/testservice": {
+				Targets: []model.LabelSet{
+					{
+						"__meta_kubernetes_service_port_protocol": "TCP",
+						"__address__":                          "testservice.own-ns.svc:30900",
+						"__meta_kubernetes_service_type":       "ClusterIP",
+						"__meta_kubernetes_service_cluster_ip": "10.0.0.1",
+						"__meta_kubernetes_service_port_name":  "testport",
+					},
+				},
+				Labels: model.LabelSet{
+					"__meta_kubernetes_service_name": "testservice",
+					"__meta_kubernetes_namespace":    "own-ns",
+				},
+				Source: "svc/own-ns/testservice",
+			},
+		},
+	}.Run(t)
+}
+
+func TestServiceDiscoveryAllNamespaces(t *testing.T) {
```

Comment:
This test makes sure that all namespace discovery is not broken by the new feature